### PR TITLE
Add `EOL`constant (End-Of-Line)

### LINKED ISSUE #10885: Add an OS-aware newline constant
Ruby has the poorly named `$/` global variable which returns `"\n"` on *nix OSes and `"\r\n"` on Windows. Crystal should have a similar constant to assist in formatting multi-line Strings in an OS-agnostic manner.

## Proposed Code

Top-level constant:

```crystal
{% if flag?(:win32) %}
  NL = "\r\n"
{% else %}
  NL = "\n"
{% end %}
```

String constant:

```crystal
class String

  {% if flag?(:win32) %}
    NL = "\r\n"
  {% else %}
    NL = "\n"
  {% end %}
  
end
```

Char vs. String (could cause issues):

```crystal
{% if flag?(:win32) %}
  NL = "\r\n"
{% else %}
  NL = '\n'
{% end %}
```

Additionally, it could be named `NEWLINE`, although `NL` is more succinct.

### Usage

```crystal
def to_s
  [
    "line one",
    "line two",
    "line three"
  ].join(NL)
end
```

### PR DESCRIPTION
**Pull request by postmodern**
* Added `EOL` to `src/kernel.cr`.
* Is OS aware (`"\r\n"` on Windows, `'\n'` otherwise).

Resolves #10885

### CODE REVIEW

**Comment by postmodern** [comment]
Some additional thoughts:
* Having `EOL` be a String on Windows, but a Char elsewhere might cause some unexpected type issues?
* Should we also add explicit CR, LF and CRLF constants? HTTP also uses `CRLF`.

**Comment by Blacksmoke16** [comment]
:+1: for `System::EOL`. I checked a few other langs and they all seem to have it in systemish namespaces:

* Java - [System.lineSeparator](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#lineSeparator%28%29) or 
* Python - [os.linesep](https://docs.python.org/3/library/os.html#os.linesep)
* C# - [Environment.NewLine](https://docs.microsoft.com/en-us/dotnet/api/system.environment.newline?view=net-5.0)

**Comment by j8r** [comment]
Am I the only one that finds `EOL`, end of line, is less correct than new line? The control character is not only used to end a file, but also separating new lines.
`EOL` is also commonly used for heredocs, which can adds to the confusion.

After reading https://en.m.wikipedia.org/wiki/Newline, there is an alternative initials `NEL`, for next line. Honestly, I'd still prefer `NEWLINE`, which is already quite small and explicit.

**Comment by j8r** [comment]
I may be alone, I still prefer `NEWLINE` (in whatever namespace it lands). Already quite small and more explicit than `EOL`.

**Comment by Sija** [comment]
@j8r If I'd like to write code that would be read by my grandma I'd agree with you for sure, but most programmers I know have a pretty firm notion of `EOF`, `EOL`, and other abbreviations from the programming world... ;)

**Comment by j8r** [comment]
Anyway, can this constant be useful in the stdlib, and even the compiler?

**Comment by HertzDevil** [comment]
https://github.com/crystal-lang/crystal/blob/4204bce75e6b79adad040f1d130adc7af95e5496/spec/std/process_spec.cr#L53-L59

**Comment by Blacksmoke16** [comment]
`System::EOL` is still my vote. Just ran into a use case where this would be helpful.

File: src/kernel.cr [hunk 1]
```diff
@@ -89,6 +89,13 @@ ARGV = Array.new(ARGC_UNSAFE - 1) { |i| String.new(ARGV_UNSAFE[1 + i]) }
 # ```
 ARGF = IO::ARGF.new(ARGV, STDIN)
 
+# The newline constant
+EOL = {% if flag?(:win32) %}
```
Comment (Blacksmoke16) [inline src/kernel.cr:93]: Given this is a windows specific thing, wouldn't the `:windows` flag make more sense here? Based on https://crystal-lang.org/reference/1.7/syntax_and_semantics/compile_time_flags.html that would make it tied to the OS versus the ABI.

**Comment by HertzDevil** [comment]
Another one:

https://github.com/crystal-lang/crystal/blob/3a360b1bfc966474c2e9f21594c7b4030d29ac73/spec/compiler/macro/macro_methods_spec.cr#L36-L42

File: src/kernel.cr (hunk 1 above)
Comment (postmodern) [reply src/kernel.cr]: Switched to checking `flag?(:windows)`. 65ee75962a7e09035af5b7525fe3a8d966610df1

**Review by straight-shoota — APPROVED** [review]
LGTM.

File: src/kernel.cr [hunk 2]
```diff
@@ -89,6 +89,13 @@ ARGV = Array.new(ARGC_UNSAFE - 1) { |i| String.new(ARGV_UNSAFE[1 + i]) }
 # ```
 ARGF = IO::ARGF.new(ARGV, STDIN)
 
+# The newline constant
+EOL = {% if flag?(:windows) %}
+        "\r\n"
+      {% else %}
+        "\n"
+      {% end %}
```
[the remark this answers is not in the document]
Comment (postmodern) [reply src/kernel.cr:97]: I find ternary statements overly confusing. They only help reduce line count at the cost of readability.

**Pull request merged**